lib/LoginStrategy.ts

```diff
@@ -255,84 +282,21 @@ export class LoginStrategy extends BaseStrategy {
     );
   }
 
-  async getRoleDefinition(namespace: string): Promise<IRoleDefinition | null> {
+  async getRoleDefinition(
+    namespace: string
+  ): Promise<IRoleDefinitionV2 | null> {
     if (this.cacheServerClient?.isAvailable) {
       return this.cacheServerClient.getRoleDefinition({ namespace });
     }
 
     const definition = await this.domainReader.read({
       node: namehash(namespace),
     });
-    return DomainReader.isRoleDefinition(definition) ? definition : null;
+    return DomainReader.isRoleDefinitionV2(definition) ? definition : null;
   }
 
-  /**
-   * @param did
-   * @returns
-   *
-   * @todo use iam-lib
-   */
-  async offchainClaimsOf(did: string): Promise<OffchainClaim[]> {
-    did = this.didUnification(did);
-
-    const transformClaim = (
-      claim: OffchainClaim
-    ): OffchainClaim | undefined => {
-      const transformedClaim = { ...claim };
-      const didFormatFields = ['iss', 'sub', 'subject', 'did', 'signer'];
-
-      let invalidDIDProperty = false;
-
-      Object.keys(transformedClaim).forEach((key) => {
-        if (didFormatFields.includes(key)) {
-          const expectedEthrDID = transformedClaim[key];
-          if (isValidErc1056(expectedEthrDID)) {
-            transformedClaim[key] = this.didUnification(transformedClaim[key]);
-          } else {
-            invalidDIDProperty = true;
-          }
-        }
-      });
-
-      return invalidDIDProperty ? undefined : transformedClaim;
-    };
-
-    const filterOutMaliciousClaims = (
-      item: OffchainClaim | undefined
-    ): item is OffchainClaim => {
-      return !!item;
-    };
-
-    if (this.cacheServerClient?.isAvailable) {
-      return (await this.cacheServerClient.getOffchainClaims({ did }))
-        .filter(isOffchainClaim)
-        .map(transformClaim)
-        .filter(filterOutMaliciousClaims);
-    }
-    const didDocument = await this.didResolver.read(did);
-    const services = didDocument.service || [];
-    return (
-      await Promise.all(
-        services.map(async ({ serviceEndpoint, ...rest }) => {
-          if (!this.isCID(serviceEndpoint)) {
```

Comment:
Do we need this `isCID` check in the `RoleCredentialResolver`?

Reply:
Hmm, right. I will add the validation in `RoleCredentialResolver`